### PR TITLE
Clippy tweaks

### PR DESCRIPTION
**Pull request by kjeremy**

### CODE REVIEW

File: crates/rust-analyzer/src/semantic_tokens.rs [hunk 1]
```diff
@@ -62,6 +62,7 @@ pub(crate) const SUPPORTED_MODIFIERS: &[SemanticTokenModifier] = &[
 #[derive(Default)]
 pub(crate) struct ModifierSet(pub(crate) u32);
 
+#[allow(clippy::suspicious_op_assign_impl)]
```
Comment (matklad) [inline crates/rust-analyzer/src/semantic_tokens.rs]: We don’t add explicit allows for clippy

File: crates/ra_ide/src/references.rs [hunk 2]
```diff
@@ -181,8 +181,8 @@ fn process_definition(
 
     for (file_id, search_range) in scope {
         let text = db.file_text(file_id);
-        let search_range =
-            search_range.unwrap_or(TextRange::offset_len(0.into(), TextUnit::of_str(&text)));
+        let search_range = search_range
+            .unwrap_or_else(|| TextRange::offset_len(0.into(), TextUnit::of_str(&text)));
```
Comment (matklad) [inline crates/ra_ide/src/references.rs]: I’d say this is a false positive, creating a range is a cheap operation

File: crates/rust-analyzer/src/semantic_tokens.rs (hunk 1 above)
Comment (kjeremy) [reply crates/rust-analyzer/src/semantic_tokens.rs]: How should this be handled then? It causes clippy to exit.

Comment (matklad) [reply crates/rust-analyzer/src/semantic_tokens.rs]: Report a bug against clippy :) and blacklist the lint in cargo xtask lint

Comment (kjeremy) [reply crates/rust-analyzer/src/semantic_tokens.rs]: Ah I missed xtask lint.

**Comment by matklad** [comment]
r=me with allow(clippy removed

**Comment by matklad** [comment]
Closing due to inactivity :)

**Pull request closed without merging**